packages/graphql-migrations/tests/specs/abstract-database.ts

```diff
@@ -155,6 +155,29 @@ test('default primary index', async () => {
   expect(id.columns).toEqual(['id'])
 })
 
+test('primary key with the @id annotation', async () => {
+  const schema = buildSchema(`
+      """
+      @model
+      """
+      type User {
+        id: ID!
```

Comment:
This is ugly probably because of https://github.com/aerogear/graphback/blob/master/packages/graphql-migrations/src/abstract/generateAbstractDatabase.ts#L149, I think we should be able to just do 
```
"""
@model
"""
type ModelWithCustomPrimaryKey {
        """
        @id
        This will be the primary key
        """
        email: String!
}
```

Reply:
Thoughts @craicoverflow

Reply:
I think we should move the check after all the fields have been parsed. But wanted to get your thoughts first.

Reply:
There is no reason to force an `id` field anymore because we map these at the resolver level.

Graphback works without an `id` field (if you are not using migrations), so there is no reason for graphql-migrations to restrict this anymore

This can be removed (whether you want to do it now or in another PR, doesn't matter :)
https://github.com/aerogear/graphback/blob/875f24585a534cedc7cf805d671c38592ea8fb53/packages/graphql-migrations/src/abstract/generateAbstractDatabase.ts#L149

Reply:
Cool, let me put that one is separate commit.